ddtelemetry/src/worker.rs

```diff
@@ -256,18 +256,15 @@ impl TelemetryWorker {
         let mut series = Vec::new();
         for (context_key, extra_tags, points) in self.data.metric_buckets.flush_series() {
             let context_guard = self.data.metric_contexts.get_context(context_key);
-            let context = match context_guard.read() {
```

Comment:
Same here. I don't understand how the fix improves anything regarding the aim of lint.
If I understand correctly, the lint is triggered when a guard is kept to long when called in as arg of a match.

It's not the case here. `context_guard` is dropped at the end of the function, because it has to live at least as long as the ref returned by `read`.
Modifying the match to an if - else doesn't  change this in the case `context_guard.read()` returns None.

Reply:
since its not a deadlock yet, I've just disabled the lint for this particular match